src/main/connect/connectionManager.ts

```diff
@@ -128,7 +129,10 @@ export class ConnectionManager implements ExtensionComponent, vscode.Disposable
 
     public async connect(): Promise<boolean> {
         this._logManager.logMessage('Trying to read credentials from Secret Storage...', 'DEBUG');
-        if (!(await this.pingCredential())) {
+        if (!(await this.loadCredential())) {
```

Comment:
When coming from "handledSignedIn", "loadCredential" is invoked twice unnecessarily